add commitRelativeChangeset config to determine what kind of changeset to commit (CNY-912)

diff -r ab3843616b90 -r 396c19713b55 conary/build/cook.py
--- a/conary/build/cook.py	Thu Nov 09 16:32:40 2006 -0500
+++ b/conary/build/cook.py	Fri Nov 10 11:40:35 2006 -0500
@@ -1109,7 +1109,10 @@ def _createPackageChangeSet(repos, db, c
         relativePackageList = []
         needTroves = {}
         for empty, p, fileMap in packageList:
-            oldTrove = previousTroveDict.get(p.getName(), None)
+            if cfg.commitRelativeChangeset:
+                oldTrove = previousTroveDict.get(p.getName(), None)
+            else:
+                oldTrove = None
             relativePackageList.append((oldTrove, p, fileMap))
 
         packageList = relativePackageList
diff -r ab3843616b90 -r 396c19713b55 conary/conarycfg.py
--- a/conary/conarycfg.py	Thu Nov 09 16:32:40 2006 -0500
+++ b/conary/conarycfg.py	Fri Nov 10 11:40:35 2006 -0500
@@ -246,6 +246,7 @@ class ConaryContext(ConfigSection):
     buildLabel            =  CfgLabel
     buildPath             =  (CfgPath, '~/conary/builds')
     cleanAfterCook        =  (CfgBool, True)
+    commitRelativeChangeset = (CfgBool, True)
     componentDirs         =  (CfgPathList, ('/etc/conary/components',
                                             '/etc/conary/distro/components',
                                             '~/.conary/components'))
